pkg/virt-handler/vm.go

```diff
@@ -187,7 +189,51 @@ func (d *VirtualMachineController) updateVMIStatus(vmi *v1.VirtualMachineInstanc
 		return err
 	}
 
-	controller.NewVirtualMachineInstanceConditionManager().CheckFailure(vmi, syncError, "Synchronizing with the Domain failed.")
+	// Update the condition when GA is connected
+	// channelState is 0 : nothing, 1: channel present, 2: channel connected
+	channelState := 0
+	channelNotPresent := true
+	if domain != nil {
+		for _, channel := range domain.Spec.Devices.Channels {
+			if channel.Target != nil {
+				log.Log.V(4).Infof("Channel: %s, %s", channel.Target.Name, channel.Target.State)
+				if channel.Target.Name == "org.qemu.guest_agent.0" {
+					channelNotPresent = false
+					if channel.Target.State == "disconnected" {
+						channelState = 1
+					}
+					if channel.Target.State == "connected" {
+						channelState = 2
+					}
+				}
+
+			}
+		}
+	}
+
+	switch {
+	case channelState == 1 && !condManager.HasCondition(vmi, v1.VirtualMachineInstanceAgentChannelOpened):
+		agentCondition := v1.VirtualMachineInstanceCondition{
+			Type:          v1.VirtualMachineInstanceAgentChannelOpened,
+			LastProbeTime: v12.Now(),
+			Status:        "Agent channel is open",
```

Comment:
Does this mean that an agent should be there, or just that we added the channel to the VMI?

Reply:
Libvirt state is as follows:

* the private channel is added and named `org.qemu.guest_agent.0`. Guest agent can communicate with the libvirt. When no agent is connected to this channel, state is set to `disconnected`
* guest agent is connected to the channel and libvirt can now pass commands. State of the channel is set to `connected`

Furthermore, when channel is present with state of disconnected, it does not say the guest agent is not present. It can be in weird state not able to connect.